### PR TITLE
395 reset id counters on new doc

### PR DESCRIPTION
**Pull request by valentindey**

### CODE REVIEW

File: app/cypress/e2e/SecvisogramPage/FormEditorTab.cy.js [hunk 1]
```diff
@@ -483,4 +483,54 @@ describe('SecvisogramPage / FormEditor Tab', function () {
       ).should('have.value', 'CSAFGID-0001')
     }
   })
+
+  it('should reset product ID counter', function () {
```
Comment (tschmidtb51) [inline app/cypress/e2e/SecvisogramPage/FormEditorTab.cy.js:487]: A similar test should be added for Group Ids...

File: app/lib/app/SecvisogramPage/View.js [hunk 2]
```diff
@@ -320,6 +324,8 @@ function View({
   }
 
   const onNewHandler = () => {
+    uniqueProductId(true)
+    uniqueGroupId(true)
```
Comment (tschmidtb51) [inline app/lib/app/SecvisogramPage/View.js:328]: This seems to be the wrong place: if I select "Cancel" in die dialogue the IDs are resetted... This should not be the case.

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Please see my comments

File: app/lib/app/SecvisogramPage/View.js (hunk 2 above)
Comment (valentindey) [reply app/lib/app/SecvisogramPage/View.js]: Indeed, I changed it and extended the test.

File: app/cypress/e2e/SecvisogramPage/FormEditorTab.cy.js (hunk 1 above)
Comment (valentindey) [reply app/cypress/e2e/SecvisogramPage/FormEditorTab.cy.js:487]: Added a test for group IDs

File: app/cypress/e2e/SecvisogramPage/FormEditorTab.cy.js [hunk 3]
```diff
@@ -483,4 +483,105 @@ describe('SecvisogramPage / FormEditor Tab', function () {
...
+  cy.get('[data-testid="new_document-create_document_button"]').click()
+  cy.get('[data-testid="alert-confirm_button"]').click()
+
+  // counter should be reset
+  cy.get(
+    '[data-testid="menu_entry-/product_tree/product_groups-add_item_button"]'
+  ).click({ force: true })
+  cy.get(
+    '[data-testid="attribute-product_tree-product_groups-0-group_id"] input'
+  ).should('have.value', 'CSAFGID-0001')
```
Comment (tschmidtb51) [inline app/cypress/e2e/SecvisogramPage/FormEditorTab.cy.js:586]: Part 2 is missing (with cancel)

File: app/lib/app/SecvisogramPage/View.js (hunk 2 above)
Comment (tschmidtb51) [reply app/lib/app/SecvisogramPage/View.js]: Thanks for extending the test. Please also extend the one for GroupIds

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Please also extend the groupID test.

File: app/lib/app/SecvisogramPage/View.js (hunk 2 above)
Comment (tschmidtb51) [reply app/lib/app/SecvisogramPage/View.js]: Does it make sense to split those tests into separate testcases? I guess it makes it cleaner...

**Review by tschmidtb51 — APPROVED** [review]
LGTM and works

**Comment by tschmidtb51** [comment]
Please make sure to close the corresponding tickets...

**Pull request merged**